packages/v8-deopt-viewer/src/index.js

```diff
@@ -110,11 +111,23 @@ export default async function run(srcFile, options) {
 	await mkdir(options.out, { recursive: true });
 
 	console.log("Parsing log...");
-	const logContents = await readFile(logFilePath, "utf8");
+
+	const fd = await openFile(logFilePath);
+	const { buffer: logContentsSlice } = await fd.read({length: 64 * 1024 * 1024});
+	await fd.close();
 
 	// New IC format has 10 values instead of 9
-	const hasNewIcFormat = /\w+IC(,.*){10}/.test(logContents);
-	const rawDeoptInfo = await parseV8Log(logContents, {
+	// todo parse first line - v8-version,8,4,371,19,-node.18,0, instead
+	const hasNewIcFormat = /\w+IC(,.*){10}/.test(logContentsSlice.toString());
+
+	// Error: Cannot create a string longer than 0x1fffffe8 characters
+	// 0x1fffffe8 = ~512 * 2 ** 20
+	// 64 * 2 ** 20 seems to be safe enough
```

Comment:
Could add what this computes to in this comment as well? If I understand correctly its about a ~67 MB string?